servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/Publisher.java

```diff
@@ -287,11 +287,54 @@ public final <R> Publisher<R> scanWith(Supplier<R> initial, BiFunction<R, ? supe
      * @param <R> Type of the items emitted by the returned {@link Publisher}.
      * @return A {@link Publisher} that transforms elements emitted by this {@link Publisher} into a different type.
      * @see <a href="https://reactivex.io/documentation/operators/scan.html">ReactiveX scan operator.</a>
+     * @deprecated Use {@link #scanWithExt(Supplier)}.
      */
+    @Deprecated
     public final <R> Publisher<R> scanWith(Supplier<? extends ScanWithMapper<? super T, ? extends R>> mapperSupplier) {
         return new ScanWithPublisher<>(this, mapperSupplier);
     }
 
+    /**
+     * Apply a function to each {@link Subscriber#onNext(Object)} emitted by this {@link Publisher} as well as
+     * optionally concat one {@link Subscriber#onNext(Object)} signal before the terminal signal is emitted downstream.
+     * <p>
+     * This method provides a data transformation in sequential programming similar to:
+     * <pre>{@code
+     *     List<R> results = ...;
+     *     ScanWithLifetimeMapperExt<T, R> mapper = mapperSupplier.get();
+     *     MappedTerminal<R> mapped = null;
+     *     try {
+     *       for (T t : resultOfThisPublisher()) {
+     *         results.add(mapper.mapOnNext(t));
+     *       }
+     *     } catch (Throwable cause) {
+     *       mapped = mapper.mapOnError(cause);
+     *       if (mapped == null) {
+     *         throw cause;
+     *       }
+     *     }
+     *     if (mapped == null) {
+     *       mapped = mapper.mapOnComplete();
+     *     }
+     *     if (mapped.onNextValid()) {
+     *       results.add(mapped.onNext());
+     *     }
+     *     if (mapped.terminal() != null) {
+     *       throw mapped.terminal();
+     *     }
+     *     return results;
+     * }</pre>
+     * @param mapperSupplier Invoked on each {@link PublisherSource#subscribe(Subscriber)} and maintains any necessary
+     * state for the mapping/accumulation for each {@link Subscriber}.
+     * @param <R> Type of the items emitted by the returned {@link Publisher}.
+     * @return A {@link Publisher} that transforms elements emitted by this {@link Publisher} into a different type.
+     * @see <a href="https://reactivex.io/documentation/operators/scan.html">ReactiveX scan operator.</a>
+     */
+    public final <R> Publisher<R> scanWithExt(
```

Comment:
Naming suggestions are welcome. Idea was to use a similar strategy to `X509TrustManager` and `X509ExtendedTrustManager` with "Ext" being short hand for "Extended".